lib/bulk/common.js

```diff
@@ -427,26 +460,560 @@ function handleMongoWriteConcernError(batch, bulkResult, ordered, err, callback)
  * @return {BulkWriteError} A BulkWriteError instance
  * @extends {MongoError}
  */
-const BulkWriteError = function(error, result) {
-  var message = error.err || error.errmsg || error.errMessage || error;
-  MongoError.call(this, message);
+class BulkWriteError extends MongoError {
+  constructor(error, result) {
+    const message = error.err || error.errmsg || error.errMessage || error;
+    super(message);
 
-  var keys = typeof error === 'object' ? Object.keys(error) : [];
-  for (var i = 0; i < keys.length; i++) {
-    this[keys[i]] = error[keys[i]];
+    Object.assign(this, error);
+
+    this.name = 'BulkWriteError';
+    this.result = result;
   }
+}
 
-  this.name = 'BulkWriteError';
-  this.result = result;
-};
-util.inherits(BulkWriteError, MongoError);
+class FindOperators {
+  constructor(self, options) {
+    this.s = Object.assign(self.s, options);
```

Comment:
This is actually assigning `options` to `self.s` and returning that. I believe what you want to do is `this.s = Object.assign({}, self.s, options);`

Reply:
Doing this causes tests to start failing and thinking that the options included in self.s are undefined. The output of doing 
"""
Object.assign({}, self.s, options)
"""
vs.
"""
Object.assign(self.s, options)
"""
is the same, but the former causes tests to fail because values are not getting filled. I'm still looking into why this would be happening, but do you have any ideas?